src/redux/actions/userDetailActions.js

```diff
@@ -1,8 +1,76 @@
-import { ADD_REVIEW } from "../constants/actionTypes";
+import { LOAD_USER_DETAIL, UPDATE_USER_DETAIL } from "../constants/actionTypes";
+import {
+  UPDATE_USER_ERROR,
+  UPDATE_USER_SUCCESS,
+  LOAD_USER_DETAILS_ERROR,
+  LOAD_USER_DETAILS_NOT_FOUND,
+} from "../constants/snackbarMessageTypes";
+import { displayError, displaySuccess } from "./snackbarActions";
+import { setLoading } from "./loadingActions";
+import axios from "axios";
 
-export const addReview = (review) => {
+const BASE_URL = `${process.env.REACT_APP_BASE_URL}/users`;
+
+const loadUserDetailSuccess = (user) => {
+  return {
+    type: LOAD_USER_DETAIL,
+    user,
+  };
+};
+
+const updateUserDetailSuccess = (user) => {
   return {
-    type: ADD_REVIEW,
-    review,
+    type: UPDATE_USER_DETAIL,
+    user,
+  };
+};
+
+export const loadUserDetails = ({ userId, currentUserId }) => {
+  return async (dispatch) => {
+    try {
+      dispatch(setLoading(true));
+      let response = await axios.get(`${BASE_URL}/${userId}/complete`);
+      let userDetails = response.data;
+      if (userDetails) {
+        let activePostingResponse =
+          currentUserId && userId === currentUserId
+            ? await axios.get(`${BASE_URL}/${userId}/postings/complete`)
```

Comment:
Note to self: This api returns the postings with the entirety of the offering objects embedded as an array of objects